src/ethernet.rs

```diff
@@ -81,84 +84,113 @@ struct VidEthertype {
     ethertype: EtherType,
 }
 
-fn to_ethertype(i: u16) -> Option<EtherType> {
```

Comment:
I removed the Option, I don't see the point here. I also implement `From` trait because this conversion can't fail.